src/geanyobject.c

```diff
@@ -380,3 +389,13 @@ GObject *geany_object_new(void)
 {
 	return g_object_new(GEANY_OBJECT_TYPE, NULL);
 }
+
+/** Get the global GeanyObject instance
+ *
+ * @return @transfer{none} The instance
+ */
+GEANY_API_SYMBOL
+GObject *geany_object_get_instance(void)
```

Comment:
BTW, couldn't we simply provide a `full` version of our custom function with the appropriate args?  Maybe not, but better be sure.